sdk/core/az_core/src/az_credential_client_secret.c

```diff
@@ -67,8 +67,13 @@ static AZ_NODISCARD az_result _az_credential_client_secret_apply(
 
   int16_t const token_length = token._internal.token_length;
 
+  // Still bad in terms of concurrency, but at least the pointer is not pointing to an unknown
+  // memory. Temporary measure to improve https://github.com/Azure/azure-sdk-for-c/issues/799 a bit.
+  uint8_t* const token_buf_ptr
```

Comment:
Can you please explain to me how this fixes the underlying issue, and maybe expand on the comment on why we still have a concurrency related bug here?

Reply:
We would get the token to a struct that is allocated on a stack, and would add a pointer to it to the http request. Once we exit, the memory is gone. Now we would initialize it with a pointer to credential's memory which is always a correct pointer, but should you read at the wrong time (a second once an hour, given that multiple threads are sending requests at the same time), you may end up reading inconsistent token - part of the old one, and part of the new one. But the memory location is correct. This is better category of a bug.